### PR TITLE
upgrade to cross-csl

### PR DESCRIPTION
**Pull request by wolverineks**

### CODE REVIEW

File: jestSetup.ts [hunk 1]
```diff
@@ -31,4 +32,4 @@ jest.mock('react-native-reanimated', () => {
   return Reanimated
 })
 jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')
-jest.mock('@emurgo/yoroi-lib-mobile', () => mockYoroiLib)
+jest.mock('@emurgo/cross-csl-mobile', () => mockCSL)
```
Comment (stackchain) [inline jestSetup.ts]: it's mocking `rnhs` already, -> csl -> rnhs (node version).

Comment (stackchain) [inline jestSetup.ts]:
```suggestion
```

File: jestSetup.ts [hunk 2]
```diff
@@ -2,9 +2,10 @@
 /* This module sets up Jest */
 import 'react-native-gesture-handler/jestSetup'
 
+import * as mockCSL from '@emurgo/cross-csl-nodejs'
```
Comment (stackchain) [inline jestSetup.ts]:
```suggestion
```

File: jestSetup.ts (hunk 1 above)
Comment (wolverineks) [reply jestSetup.ts]: where is that happening?

Comment (stackchain) [reply jestSetup.ts]: > where is that happening?

inside `@__mocks__` (perhaps we should move all inside setup dunno)

File: src/yoroi-wallets/Wallet.ts [hunk 3]
```diff
@@ -14,8 +13,8 @@ import {Logger} from '../legacy/logging'
 import {getCardanoNetworkConfigById, isJormungandr} from '../legacy/networks'
 import {IsLockedError, nonblockingSynchronize, synchronize} from '../legacy/promise'
 import type {BackendConfig, EncryptionMethod} from '../legacy/types'
-import {NetworkId, WalletImplementationId, YoroiProvider} from './cardano'
-import {AddressChain, AddressChainJSON, Addresses} from './cardano/chain'
+import {CardanoTypes, NetworkId, WalletImplementationId, YoroiProvider} from './cardano'
+import {AddressChain, AddressChain, AddressChainJSON, AddressChainJSON, Addresses} from './cardano/chain'
```
Comment (stackchain) [inline src/yoroi-wallets/Wallet.ts]:
```suggestion
import {AddressChain, AddressChainJSON, Addresses} from './cardano/chain'
```

**Review by stackchain — CHANGES_REQUESTED** [review]
can we test mocking only rnhs

**Review by stackchain — APPROVED** [review]

**Pull request merged**